src/routes/layout.tsx

```diff
@@ -62,31 +67,37 @@ export default component$(() => {
   const wagmiConfig = useContext(WagmiConfigContext);
   const login = useContext(LoginContext);
   // eslint-disable-next-line qwik/no-use-visible-task
-  useVisibleTask$(() => {
+  useVisibleTask$(async () => {
     const wconfig = defaultWagmiConfig({
       chains: [mainnet, sepolia],
       projectId: import.meta.env.PUBLIC_PROJECT_ID,
       metadata,
     });
 
     wagmiConfig.config = noSerialize(wconfig);
-
     if (wagmiConfig.config) {
       watchAccount(wagmiConfig.config!, {
         onChange(account) {
-          if (window.location.pathname === "/") {
+          if (
+            window.location.pathname === "/signin" ||
+            window.location.pathname === "/"
+          ) {
             localStorage.setItem(
               "emmethUserWalletAddress",
               `${account.address}`,
             );
-            login.account = noSerialize(account);
-            login.address.value = account.address;
-            login.chainId.value = account.chainId;
           } else {
             reconnect(wagmiConfig.config as Config);
           }
+          login.account = noSerialize(account);
+          login.address.value = account.address;
+          login.chainId.value = account.chainId;
         },
       });
+      console.log(
+        "[connections]: ",
+        await getConnections(wagmiConfig.config as Config),
+      );
```

Comment:
I guess we could remove async from this `visibleTask$` after getting rid of `console.log()` with `getConnections(...)`